contributors/design-proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,315 @@
+  Logs *DiskUsage `json:"logs,omitempty"`  
+}  
+
+// CpuUsage holds statistics about the amount of cpu time consumed  
+type CpuUsage struct {  
+  // The time at which these stats were updated.  
+  Timestamp metav1.Time `json:"time"`  
+  // Total CPU usage (sum of all cores) averaged over the sample window.  
+  // The "core" unit can be interpreted as CPU core-nanoseconds per second.  
+  // +optional  
+  UsageNanoCores *uint64 `json:"usageNanoCores,omitempty"`  
+  // Cumulative CPU usage (sum of all cores) since object creation.  
+  // +optional  
+  UsageCoreNanoSeconds *uint64 `json:"usageCoreNanoSeconds,omitempty"`  
+}  
+
+// MemoryUsage holds statistics about the quantity of memory consumed  
+type MemoryUsage struct {  
+  // The time at which these stats were updated.  
+  Timestamp metav1.Time `json:"time"`  
+  // The amount of working set memory. This includes recently accessed memory,  
+  // dirty memory, and kernel memory.  
+  // +optional  
+  WorkingSetBytes *uint64 `json:"workingSetBytes,omitempty"`  
+}  
+
+// VolumeUsage holds statistics about the quantity of disk resources consumed for a volume  
+type VolumeUsage struct {  
+  // Embedded DiskUsage  
+  DiskUsage  
+  // Name is the name given to the Volume  
+  // +optional  
+  Name string `json:"name,omitempty"`  
+}  
+
+// DiskUsage holds statistics about the quantity of disk resources consumed  
+type DiskUsage struct {  
+  // The time at which these stats were updated.  
+  Timestamp metav1.Time `json:"time"`  
+  // The device on which resources are consumed  
+  Device string `json:"device"`  
+  // UsedBytes represents the disk space consumed on the device, in bytes.  
+  // +optional  
+  UsedBytes *uint64 `json:"usedBytes,omitempty"`  
+  // InodesUsed represents the inodes consumed on the device  
+  // +optional  
+  InodesUsed *uint64 `json:"inodesUsed,omitempty"`  
+}  
+```
+
+### Core Machine Info:
+
+In addition to providing metrics, cAdvisor also provides machine info.  While it is not neccessary to use this structure for reporting Machine Info, the following contains all of the information provided to the kubelet by cAdvisor, generally used at startup.
+
+The code that provides this data currently resides in cAdvisor.  I propose moving this to the kubelet as well.
+
+```go
+type CoreInfo struct {  
+  // MachineID reported by the node. For unique machine identification  
+  // in the cluster this field is prefered. Learn more from man(5)  
+  // machine-id: http://man7.org/linux/man-pages/man5/machine-id.5.html  
+  MachineID string `json:"machineID" protobuf:"bytes,1,opt,name=machineID"`  
+  // SystemUUID reported by the node. For unique machine identification  
+  // MachineID is prefered. This field is specific to Red Hat hosts  
+  // https://access.redhat.com/documentation/en-US/Red_Hat_Subscription_Management/1/html/RHSM/getting-system-uuid.html  
+  SystemUUID string `json:"systemUUID" protobuf:"bytes,2,opt,name=systemUUID"`  
```

Comment:
IIRC, @dchen1107 and @philips had intended to drop `MachineID` in favor of `SystemUUID` or vice versa. Now might be the time to do that.